include/arch/arm/irq.h

```diff
@@ -99,20 +98,25 @@ extern void _arch_isr_direct_pm(void);
 #endif
 
 #define ARCH_ISR_DIRECT_HEADER() arch_isr_direct_header()
-extern void arch_isr_direct_header(void);
-
 #define ARCH_ISR_DIRECT_FOOTER(swap) arch_isr_direct_footer(swap)
 
 /* arch/arm/core/exc_exit.S */
 extern void z_arm_int_exit(void);
 
 #ifdef CONFIG_TRACING
+extern void sys_trace_isr_enter(void);
 extern void sys_trace_isr_exit(void);
 #endif
 
-static inline void arch_isr_direct_footer(int maybe_swap)
+static inline void arch_isr_direct_header(void)
 {
+#ifdef CONFIG_TRACING
```

Comment:
@stephanosio are the #ifdefs here needed? Not present in the original place where you moved this function from

Reply:
That's cleaner, and consistent with what we do in the _footer function